File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Action/Common/EscapeAction.ts
import Async from "../../../../../Core/Async";
import { AbstractMonster } from "../../Unit/Monster/AbstractMonster";
import AbstractGameAction, { ActionType } from "../AbstractGameAction";

export class EscapeAction extends AbstractGameAction{
    Type = ActionType.TEXT
    public constructor(private enemy:AbstractMonster) {
        super();
    }
    
    async Update(): Promise<boolean> {
        if(this.Duration == 0){
            this.enemy.Escape();
        }
        await Async.Delay(500)
        return super.Update();
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Action/Combat/EscapeAction.cs
using Cysharp.Threading.Tasks;
using ModTheSpire.Event;
using ModTheSpire.Unit.Monster;

namespace ModTheSpire.Effect.Other
{
    public class EscapeAction : IGameAction
    {
        private readonly AbstractMonster mMonster;
        public float Duration { get; set; }
        public ActionType Type { get; } = ActionType.TEXT;
        public void Init()
        {
        }

        public EscapeAction(AbstractMonster monster)
        {
            mMonster = monster;
        }
        
        public async UniTask<bool> Update()
        {
            mMonster.Escape();
            await UniTask.Delay(500);
            return true;
        }
    }
}